app/src/main/java/io/seqera/wave/cli/Client.java

```diff
@@ -194,24 +196,41 @@ protected URI imageToManifestUri(String image) {
         return URI.create(result);
     }
 
-    protected void awaitImage(String image) {
-        final URI manifest = imageToManifestUri(image);
+    void awaitCompletion(String buildId, String imageName) {
+        final long maxAwait = Duration.ofMinutes(15).toMillis();
+        final long startTime = Instant.now().toEpochMilli();
+        while( true ) {
+            if( isComplete(buildId) ) {
+                return;
+            }
+            if( System.currentTimeMillis()-startTime > maxAwait ) {
+                break;
+            }
+        }
+    }
+
+    protected boolean isComplete(String buildId) {
+        final String statusEndpoint = endpoint + "/v1alpha1/builds/"+buildId+"/status";
         final HttpRequest req = HttpRequest.newBuilder()
-                .uri(manifest)
-                .headers(REQUEST_HEADERS)
-                .timeout(Duration.ofMinutes(5))
+                .uri(URI.create(statusEndpoint))
+                .headers("Content-Type","application/json")
                 .GET()
                 .build();
-        final long begin = System.currentTimeMillis();
-        final HttpResponse<String> resp = httpSend(req);
-        final int code = resp.statusCode();
-        if( code>=200 && code<400 ) {
-            final long delta = System.currentTimeMillis()-begin;
-            log.debug("Wave container available in {} [{}] {}", delta, code, resp.body());
+
+        try {
+            final HttpResponse<String> resp = httpSend(req);
```

Comment:
Shouldn't we add a delay before making another request?
 it will create a lot of HTTP requests to wave

Reply:
it make 475 calls for this command
`%wave -s --platform arm64 --freeze --wave-endpoint https://00ffbc112673.ngrok.app --conda-package salmon    --conda-package Jedi --tower-token "eyJ0aWQiOiAyfS5jNDk0MjhlNjAwNWViMzI2OGRmZGM0Mjg5MzlmYzI5M2ZjNTFmZDQz" --tower-endpoint http://localhost:8008/api --build-repo "docker.io/hrma017/dev" --await`

Reply:
@pditommaso thoughts?

Reply:
YES

Reply:
10 secs delay should be good

Reply:
ok i will add it

Reply:
done